### DISCUSSION TITLE
WebView support page zoom

### DISCUSSION THREAD

**Discussion by rexzhang** [beeware/toga#2328] [Ideas]
### What is the problem or limitation you are having?

I'm trying to implement a program for mac OS/ Windows/ Linux based mostly on Web View using toga+briefcase. I want the user to be able to control the page scaling by themselves, what about like the `CMD+=` in browser.
I tried to get the page content and change `<meta name="viewport" content="width=device-width,initial-scale=1.0" />` to `<meta name="viewport" content="width=device-width,initial-scale=2.0" />`, but it doesn't work!

### Describe the solution you'd like

WebView have one property: page_zoom, we can get/set it. like [this](https://developer.apple.com/documentation/webkit/wkwebview/3516411-pagezoom) 

### Describe alternatives you've considered

I have no idea.

### Additional context

more function for WebView, like:
- get_content, Corresponds to set_content. It will be WebView that becomes more powerful
- go_back or get_back_forward_list. will be useful
- get/set cookies
- get/set local storage
- get/set session storage

**Comment by freakboy3742** [comment]
Converting to a discussion since this isn't a single resolvable idea in itself.

WebView isn't a fully fledged browser implementation - it's a HTML canvas. If you want your app to respond to CMD + or CMD - zoom commands, then your app should implement command handlers to perform those operations. To that end, it doesn't make any sense for Toga to have default zoom keyboard shortcuts.

If you've got a specific bug to report around zoom handling, you'll need to provide a complete bug report detailing what you've tried, the platforms you've tried on, and what behavior you expected to see.

As for your other "more function" suggestions - a ticket should contain a single resolvable idea, so that when that feature is implemented, the ticket can be closed. However, the initial guidance I gave in this reply applies - WebView *isn't* a browser. It is a primitive with which you could potentially *build* a browser. As a result, we're not necessarily looking to add lots of "browser like" behaviour to the WebView API. Some of the APIs you've suggested might be plausible to add as first-class functions - however, they're also, for the most part, things that could be implemented using the existing `execute_javascript()` API.
